### PR TITLE
dssim: with-library

### PR DESCRIPTION
**Pull request by s172262**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
----
```
$ brew info dssim
dssim: stable 1.3.2 (bottled)
RGBA Structural Similarity C implementation (with a Rust API)
https://github.com/pornel/dssim
/usr/local/Cellar/dssim/1.3.2 (9 files, 97.3KB) *
  Built from source on 2017-10-19 at 20:53:00 with: --with-library
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/dssim.rb
==> Dependencies
Build: pkg-config ✔
Required: libpng ✔
==> Options
--with-library
	Build library (require `python3` and `ninja`)

$ brew list dssim --verbose
/usr/local/Cellar/dssim/1.3.2/INSTALL_RECEIPT.json
/usr/local/Cellar/dssim/1.3.2/LICENSE
/usr/local/Cellar/dssim/1.3.2/bin/dssim
/usr/local/Cellar/dssim/1.3.2/.brew/dssim.rb
/usr/local/Cellar/dssim/1.3.2/include/dssim.h
/usr/local/Cellar/dssim/1.3.2/README.md
/usr/local/Cellar/dssim/1.3.2/lib/pkgconfig/dssim.pc
/usr/local/Cellar/dssim/1.3.2/lib/libdssim-lib.dylib
/usr/local/Cellar/dssim/1.3.2/lib/libdssim-lib.dylib.1.1

$ otool -L /usr/local/Cellar/dssim/1.3.2/bin/dssim
/usr/local/Cellar/dssim/1.3.2/bin/dssim:
	/usr/local/Cellar/dssim/1.3.2/lib/libdssim-lib.dylib.1.1 (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/opt/libpng/lib/libpng16.16.dylib (compatibility version 51.0.0, current version 51.0.0)
	/usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.11)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.0.0)
```

### CODE REVIEW

File: Formula/dssim.rb [hunk 1]
```diff
@@ -13,12 +13,37 @@ class Dssim < Formula
     sha256 "51756e74240d03c87a79ff0e14e494d249701d4069ea336b529b47a179c469d3" => :mavericks
   end
 
+  option "with-library", "Build library (require `python3` and `ninja`)"
+
+  if build.with? "library"
+    resource "meson-0.27.0" do
```
Comment (ilovezfs) [inline Formula/dssim.rb]: we have a meson formula

**Comment by ilovezfs** [comment]
I would just build this unconditionally without an option.

File: Formula/dssim.rb (hunk 1 above)
Comment (s172262) [reply Formula/dssim.rb]: Building `dssim-1.3.2` needs just `meson-0.27.0`.
`meson-0.28.0` and newer was removed `pkbconfig_gen()` function which was used in "meson.build" of `dssim-1.3.2`.

Comment (ilovezfs) [reply Formula/dssim.rb]: They'll need to fix that then.

Comment (s172262) [reply Formula/dssim.rb]: I will try offical patch as soon as possible.

Comment (ilovezfs) [reply Formula/dssim.rb]: thanks :)

**Comment by s172262** [comment]
We might wait for version 1.3.2.1 of `dssim`.

**Pull request closed without merging**